### PR TITLE
Feature/GSYE-15: Open the Smart Meter asset to external connection

### PR DESCRIPTION
**Pull request by mrsaemir**
## Reason for the proposed changes
As requested by [GSYE-15](https://gridsingularity.atlassian.net/browse/GSYE-15): Open the Smart Meter asset to external connection

## Proposed changes
- Implement `SmartMeterExternalStrategy`
- Write tests for `SmartMeterExternalStrategy`
- Small refactors to other external strategies

## Related PRs
- [GBIT](https://github.com/gridsingularity/gsy-backend-integration-tests/pull/60)

**INTEGRATION_TESTS_BRANCH**=feature/GSYE-15
**GSY_FRAMEWORK_BRANCH**=master

### CODE REVIEW

File: src/gsy_e/models/strategy/external_strategies/smart_meter.py [hunk 1]
```diff
@@ -0,0 +1,438 @@
+# pylint: disable=broad-except
```
Comment (hannesdiedrich) [inline src/gsy_e/models/strategy/external_strategies/smart_meter.py:1]: We try not to disable pylint warnings on module basis. If you see occurrences, these are from the time before we discussed that this is not a good idea, please remove then, thanks!

**Review by hannesdiedrich — COMMENTED** [review]
Thanks a lot for this PR!
I propose to add an integration test to the 
gsy-backend-integration-tests.gsy_e_sdk_tests in order to add a real end-to-end test.
As the SmartMeter has the same functionalities as the Storage (posting bids AND offers) the test setup for the storage could be helpful for your understanding: https://github.com/gridsingularity/gsy-backend-integration-tests/blob/master/gsy_e_sdk_tests/integration_tests/test_aggregator_ess.py 
Let me know if you need help with adding integration tests. Thanks a lot!

**Review by hannesdiedrich — APPROVED** [review]
LGTM

**Review by spyrostz — APPROVED** [review]
LGTM

**Pull request merged**